karpenter/add-karpenter-donot-disrupt/add-karpenter-donot-disrupt.yaml

```diff
@@ -0,0 +1,34 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: add-karpenter-donot-disrupt
+  annotations:
+    policies.kyverno.io/title: Add Karpenter Do Not Disrupt
+    policies.kyverno.io/category: Karpenter, EKS Best Practices
+    policies.kyverno.io/severity: medium
+    policies.kyverno.io/subject: Pod
+    kyverno.io/kyverno-version: 1.7.1
+    policies.kyverno.io/minversion: 1.6.0
+    kyverno.io/kubernetes-version: "1.23"
+    policies.kyverno.io/description: >-
+      This policy assists in migrating from Karpenter 0.x to 1.x by adding the
+      karpenter.sh/do-not-disrupt label to pods that have the deprecated
+      karpenter.sh/do-not-evict label. This ensures smooth upgrades and
+      maintains the intended protection against pod disruption.
+spec:
+  rules:
+    - name: add-donot-disrupt-label
+      match:
+        resources:
+          kinds:
+            - Pod
+      preconditions:
+        all:
+          - key: '{{ request.object.metadata.labels."karpenter.sh/do-not-evict" || '''' }}'
+            operator: NotEquals
+            value: ""
```

Comment:
Is this condition needed? We have `+()` to add-if-not-present.

Reply:
The precondition ensures that the new `do-not-disrupt` label is added only when the old `do-not-evict` label is present and set to `"true."` This way, it is only migrating resources explicitly marked for non-eviction in Karpenter 0.x.
 
  This `+()` could add the label to all resources, but aiming for a targeted migration, so verifying the old label first ensuring only the intended resources are affected.

Reply:
I missed it, it's not the key :(